src/textures/utils/coordinate-transform.ts

```diff
@@ -0,0 +1,115 @@
+import { ringToPoints } from "../../geoms/brep-converter"
+
+export interface CoordinateTransform {
+  canvasXFromPcb: (x: number) => number
+  canvasYFromPcb: (y: number) => number
+}
+
+/**
+ * Create coordinate transformation functions for PCB to canvas mapping
+ */
+export function createCoordinateTransform(
+  boardOutlineBounds: {
+    minX: number
+    maxX: number
+    minY: number
+    maxY: number
+  },
+  traceTextureResolution: number,
+): CoordinateTransform {
+  const canvasXFromPcb = (pcbX: number) =>
+    (pcbX - boardOutlineBounds.minX) * traceTextureResolution
+  const canvasYFromPcb = (pcbY: number) =>
+    (boardOutlineBounds.maxY - pcbY) * traceTextureResolution
+
+  return { canvasXFromPcb, canvasYFromPcb }
+}
+
+/**
+ * Set up canvas context for bottom layer rendering
+ */
+export function setupBottomLayerContext(
+  ctx: CanvasRenderingContext2D,
+  canvasHeight: number,
+): void {
+  ctx.translate(0, canvasHeight)
+  ctx.scale(1, -1)
+}
+
+/**
+ * Draw a polygon shape on canvas
+ */
+export function drawPolygon({
+  ctx,
+  points,
+  canvasXFromPcb,
+  canvasYFromPcb,
+}: {
+  ctx: CanvasRenderingContext2D
+  points: [number, number][]
+  canvasXFromPcb: (x: number) => number
+  canvasYFromPcb: (y: number) => number
+}): void {
+  if (points.length < 3) return
+
+  ctx.beginPath()
+  points.forEach((point, index) => {
+    const canvasX = canvasXFromPcb(point[0])
+    const canvasY = canvasYFromPcb(point[1])
+    if (index === 0) {
+      ctx.moveTo(canvasX, canvasY)
+    } else {
+      ctx.lineTo(canvasX, canvasY)
+    }
+  })
+  ctx.closePath()
+  ctx.fill()
+}
+
+/**
+ * Draw brep shape using custom implementation
+ */
+export function drawBrepShape({
```

Comment:
why is all of this being re-implemented all the time, just implement in circuit-to-canvas, people are just repeating code and creating bugs everywhere when they repeat common functionality like this, circuit-to-canvas should have all the stuff for drawing elements, why implement something custom